java/casting.md

```diff
@@ -8,4 +8,10 @@ Mit einem cast lässt sich nicht jedes Objekt X in ein Objekt Y umwandeln. Es ka
 
 Beispielsweise kann man einen `CommandSender` zu `Player` casten, weil `Player` von `CommandSender` erbt. Es gibt aber noch andere `CommandSender` - z.B. die Konsole oder Command Blocks - deshalb sollte man vorher mit `instanceof` überprüfen, ob es sich wirklich um einen `Player` handelt.
 
+Das in Java 14 eingeführte `Pattern Matching` ist dabei sehr nützlich - das führt zu simplerem und einfacherem Code.
+```java
+if (sender instanceof Player player) 
```

Comment:
Also, this doesn't incite to use early return, as users see only how to check for it, and think that it only works inside if clauses.

Reply:
We should probably also include that stuff like this works, since it actively encourages a better code style:
```java
if(!(sender instanceof Player player)) return;
player.setGameMode(GameMode.CREATIVE);
```

Reply:
We could also add another tag for explicit pattern and use the related tags feature once it is out https://github.com/rainbowdashlabs/krile/issues/22

Or we just add a more detailed description and examples on a second page

Reply:
Usually I do exactly the code like you did. I think it is to difficult to Understand Pattern Matching with this small example. We could link this: https://www.baeldung.com/regular-expressions-java

Reply:
And also show usage with concated conditions.

```java
if(sender instanceof Player player && player.hasPermission("toggle.creative")){
    player.setGameMode(GameMode.CREATIVE);
}
```

Reply:
Thats why I suggest a second page or tag to describe it more detailed. I also think you linked the wrong page xD Usually linking to an english source doesnt really help most of the beginners. They simply stop looking at it once they see english text.

Reply:
Whoopsi: https://www.baeldung.com/java-pattern-matching-instanceof

Reply:
> Thats why I suggest a second page or tag to describe it more detailed. I also think you linked the wrong page xD Usually linking to an english source doesnt really help most of the beginners. They simply stop looking at it once they see english text.

I guess another tag for this would be better, something like: "Für eine einfache Umsetzung, schau dir /tag instanceof-pattern-machting an."

Reply:
> Thats why I suggest a second page or tag to describe it more detailed. I also think you linked the wrong page xD Usually linking to an english source doesnt really help most of the beginners. They simply stop looking at it once they see english text.

We should add multiple pages with an explanation foreach. Which advantages they have.